pkg/dashboard/functiontemplates/generatedfunctiontemplatefetcher.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functiontemplates
+
+import (
+	"encoding/base64"
+
+	"github.com/nuclio/nuclio/pkg/errors"
+
+	"github.com/ghodss/yaml"
+)
+
+type GeneratedFunctionTemplateFetcher struct {
+	functionTemplates []*FunctionTemplate
+	FunctionTemplateFetcher
+}
+
+func NewGeneratedFunctionTemplateFetcher() (*GeneratedFunctionTemplateFetcher, error) {
+	generatedFunctionTemplates := GeneratedFunctionTemplates
+
+	// populate encoded field of templates so that when we are queried we have this ready
+	if err := enrichFunctionTemplates(generatedFunctionTemplates); err != nil {
+		return nil, errors.Wrap(err, "Failed to populated serialized templates")
+	}
+
+	functionTemplatesFromGeneratedFunctionTemplates, err := generatedFunctionTemplatesToFunctionTemplates(generatedFunctionTemplates)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to generate regular functionTemplates out og generatedFunctionTemplates")
+	}
+
+	return &GeneratedFunctionTemplateFetcher{
+		functionTemplates: functionTemplatesFromGeneratedFunctionTemplates,
+	}, nil
+}
+
+func NewGeneratedFunctionTemplateFetcherFromTemplates(generatedFunctionTemplates []*generatedFunctionTemplate) (*GeneratedFunctionTemplateFetcher, error) {
+
+	// populate encoded field of templates so that when we are queried we have this ready
+	if err := enrichFunctionTemplates(generatedFunctionTemplates); err != nil {
+		return nil, errors.Wrap(err, "Failed to populated serialized templates")
+	}
+
+	functionTemplatesFromGeneratedFunctionTemplates, err := generatedFunctionTemplatesToFunctionTemplates(generatedFunctionTemplates)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to generate regular functionTemplates out og generatedFunctionTemplates")
+	}
+
+	return &GeneratedFunctionTemplateFetcher{
+		functionTemplates: functionTemplatesFromGeneratedFunctionTemplates,
+	}, nil
+}
+
+func (gftf *GeneratedFunctionTemplateFetcher) Fetch() ([]FunctionTemplate, error) {
```

Comment:
Don't speak in two terms. Either use []FunctionTemplate everywhere or []*FunctionTemplate. I prefer the latter.
It's possible to iterate over a list of concrete structs without copy, it's just annoying.

Also, next time don't do a make with len. Just use append - it'll grow automatically. It's OK for slow path